tensorflow/stream_executor/dnn.h

```diff
@@ -826,6 +826,8 @@ class AlgorithmConfig {
  public:
   AlgorithmConfig() {}
   explicit AlgorithmConfig(AlgorithmDesc algorithm) : algorithm_(algorithm) {}
+  explicit AlgorithmConfig(AlgorithmDesc algorithm, size_t scratch_size)
```

Comment:
No 'explicit'.

Could you please document how scratch_size is used and whether/how it differs between CUDA and ROCm?

Would it make sense for the CUDA path to report the required scratch size as well?

Reply:
@chsigg. I've removed `explicit` from this ctor. Also I put more comments to explain the rationale behind this `scratch_size` field.


Here's an more elaborated explaination from @deven-amd 

That need is primarily due to a difference between the cudnn/miopen apis on the cudnn side, there is a API ( cudnnGetConv*WorkspaceSize ) which will return the scratch_size needed for a given conv parameters tuple (algo, conv shape).  This API is called during each convolution call to determine the required scratch size.

On the miopen side, there does not exist such an API, and therefore things need to be done a little differently. During the auto-tuning phase, we know the scratch_size required by the best algorithm (corresponding to a given conv parameter tuple), and we need to store this information for subsequent use. That is the reason why we need to add the scratch_size field to the AlgorithmConfig

This also means that the scratch_size field in AlgorithmConfig is populated and used in ROCm only (since there is no need to do so on the CUDA side).